### PR TITLE
First steps towards emitting a mildly compact path

### PR DESCRIPTION
**Pull request by rsheeter**
Start of #18, add optional icon path optimization, implement use of line shorthands and relative coords when shorter.

### CODE REVIEW

**Review by EsoFromTheEast — APPROVED** [review]

File: src/pathstyle.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    if relative.as_ref().map(|s| s.len()).unwrap_or(usize::MAX) < absolute.len() {
+        svg.push(prefix.to_ascii_lowercase());
+        svg.push_str(&relative.unwrap());
+    } else {
+        svg.push(prefix);
+        svg.push_str(&absolute);
+    }
+}
+
+fn to_unoptimized_svg_path(path: &BezPath) -> String {
```
Comment (dfrg) [inline src/pathstyle.rs:87]: I’m not sure if you’re planning future optimizations that will make merging more unpleasant but in the current state, it looks like the two svg path functions can be nicely combined.

File: src/pens.rs [hunk 2]
```diff
@@ -1,73 +1,27 @@
...
         Self {
-            transform: Affine::FLIP_Y, // svg is Y-down, fonts are Y-up
             path: Default::default(),
         }
     }
 
     fn to_svg_units(&self, x: f32, y: f32) -> Point {
-        self.transform * Point::new(x as f64, y as f64)
+        // svg is Y-down, fonts are Y-up
+        Affine::FLIP_Y * Point::new(x as f64, y as f64)
```
Comment (dfrg) [inline src/pens.rs:20]: If we’re not doing arbitrary transforms, this should be equivalent to just negating the y coord.

**Review by dfrg — APPROVED** [review]
LGTM with a couple suggestions.

File: src/pens.rs (hunk 2 above)
Comment (rsheeter) [reply src/pens.rs:20]: We might add back more than flip but for now it can just be -y indeed. Good catch.

**Pull request merged**